include/aspect/introspection.h

```diff
@@ -29,10 +29,289 @@
 
 #include <aspect/parameters.h>
 
+namespace dealii
+{
+  /**
+   * Structure to describe everything necessary to define a single variable of the class
+   * FEVariableSystem. It groups the FiniteElement<dim> with other information like a name and the block structure
+   * used in linear systems. A vector of these structs will be converted into
+   * an ordered collection of objects of type an FEVariable represented by FEVariableSystem.
+   *
+   * Blocks are used to extract or solve for certain variables separately or in block based preconditioners.
+   * As an example, in a Stokes problem one might want velocity and pressure in a single block (for use with a direct solver),
+   * or one block for all velocity components and a second block for the pressure (for a Schur complement-based preconditioner),
+   * or separate blocks for pressure and each velocity component.
+   */
+  template <int dim>
+  struct VariableDefinition
+  {
+    /**
+     * Constructor.
+     *
+     * @param name A user-friendly and unique string representation of the variable.
+     *
+     * @param fe The FiniteElement class to use.
+     *
+     * @param multiplicity Number of copies of the @p fe to create.
+     *
+     * @param n_blocks Number of blocks this variable represents inside the
+     * linear system. A value of 0 will add the next variable to the current
+     * block, while 1 will put the next variable into a new block. A value that is
+     * equal to the number of components will create a block for each
+     * component.
+     */
+    VariableDefinition(const std::string &name,
+                       std_cxx11::shared_ptr<FiniteElement<dim> > fe,
+                       const unsigned int multiplicity,
+                       const unsigned int n_blocks)
+      : name(name),
+        fe(fe),
+        multiplicity(multiplicity),
+        n_blocks(n_blocks)
+    {
+      Assert(n_blocks == 0
+             || n_blocks == 1
+             || n_blocks == n_components(),
+             ExcMessage("A Variable can only have 0, 1, or n_components() number of blocks"));
+    }
+
+    /**
+     * Return the total number of components of this variable.
+     */
+    unsigned int n_components() const
+    {
+      return fe->n_components() * multiplicity;
+    }
+
+    std::string name;
+    std_cxx11::shared_ptr<FiniteElement<dim> > fe;
+    unsigned int multiplicity;
+    unsigned int n_blocks;
```

Comment:
My vote would be to keep it as is. If I can remove an entry and add an entry I can also indirectly modify an entry, right? So I don't mind that we can edit things directly. Maybe you want to change the pressure degree for example...